### ISSUE TITLE
Meshes are not visualized properly

### ISSUE THREAD

**Issue by kongsgard** [intel-isl/Open3D#1096]
**Describe the bug**
Meshes are not visualized properly, as there is some clipping and missing triangles.

**To Reproduce**
```
# Load e.g. knot.ply in TestData
mesh = o3d.io.read_triangle_mesh("knot.ply")
o3d.visualization.draw_geometries([mesh])
```
Showing the wireframe also highlights the issues.

**Expected behavior**
Visualize the mesh as in Open3D 0.7.0.

**Screenshots**
<img width="1054" alt="Screenshot 2019-07-30 at 19 15 25" src="https://user-images.githubusercontent.com/13556755/62150493-69eba900-b2fe-11e9-8e50-3b606e739e0f.png">

**Environment:**
 - OS: macOS Mojave
 - Python version: 3.6
 - Open3D version: -
 - Is this remote workstation?:  no
 - How did you install Open3D?: build_from_source

**Additional context**
I tested previous commits, and it seems like the bug was introduced in #1080.

**Comment by griegler** [comment]
Seems to be a regression of #1080. I will have a look at it.

**Comment by griegler** [comment]
Fixed in #1122
